### DISCUSSION TITLE
为什么等于contained==1匹配满足条件，等于2的时候同样也有交集啊，怎么不算呢

### DISCUSSION THREAD

**Discussion by cqray1990** [RapidAI/TableStructureRec#67] [Q&A]
contained = is_box_contained(ocr_box, pred_box, 0.6)
            if contained == 1 or calculate_iou(ocr_box, pred_box) > 0.8:
                if j not in matched:
                    matched[j] = [gt_box]
                else:
                    matched[j].append(gt_box)
            else:
                not_match_orc_boxes.append(gt_box)

**Comment by Joker1212** [comment]
因为1标识第一个box被包含，而表格的box在结构恢复时，是会比ocr的box更大的，ocr的box是不会换行，且同行也可能分块的

**Comment by Joker1212** [comment]
这里恢复表格的思路是将ocr的box放在合适的表格的box里面

**Reply by cqray1990** [reply]
![box_in_table](https://github.com/user-attachments/assets/49329f0f-8afb-467d-8b6d-b869d6456fbc)
[>)) 这里恢复表格的思路是将ocr的box放在合适的表格的box里面
上面结果是ocr文本结果第一行就会被过滤掉，导致检测到的表格box没有匹配到，也被过滤掉了
但是如果ocr文本的框比表格的box大就会导致表格的box被忽略掉，同时ocrtext也是被过滤掉了，ocr的框包含了 表格的框

**Reply by Joker1212** [reply]
不是过大会被过滤，只要阈值符合就行了，其实条件很宽泛，大概非重叠区域占比40%都算匹配到了，这种场景其实就是ocr框太大了，emmm,最近在研究单字ocr的方案，应该可以解决这种问题
